app/javascript/packs/braintree3DSecureCustomerForm.js

```diff
@@ -0,0 +1,112 @@
+import client from 'braintree-web/client'
+import hostedFields from 'braintree-web/hosted-fields'
+import threeDSecure from 'braintree-web/three-d-secure'
+
+const form = document.querySelector('#customer_form')
+const clientToken = form.dataset.clientToken
+const braintreeNonce = document.querySelector('#braintree_nonce')
+
+const billingAdressInfo = {
+  company: document.querySelector('#customer_credit_card_billing_address_company').value,
+  streetAddress: document.querySelector('#customer_credit_card_billing_address_street_address').value,
+  postalCode: document.querySelector('#customer_credit_card_billing_address_postal_code').value,
+  locality: document.querySelector('#customer_credit_card_billing_address_locality').value,
+  region: document.querySelector('#customer_credit_card_billing_address_region').value,
+  countryCodeAlpha2: document.querySelector('#customer_credit_card_billing_address_country_name').value
+}
+
+const hostedFieldOptions = {
+  styles: {
+    'input': {
+      'font-size': '14px'
+    },
+    'input.invalid': {
+      'color': 'red'
+    },
+    'input.valid': {
+      'color': 'green'
+    }
+  },
+  fields: {
+    number: {
+      selector: '#customer_credit_card_number',
+      placeholder: '4111 1111 1111 1111'
+    },
+    cvv: {
+      selector: '#customer_credit_card_cvv',
+      placeholder: '123'
+    },
+    expirationDate: {
+      selector: '#customer_credit_card_expiration_date',
+      placeholder: 'MM/YY'
+    }
+  }
+}
+
+const threeDSecureParameters = {
+  amount: '00.00',
+  billingAddress: billingAdressInfo,
```

Comment:
It could be useful to add the email of the cardholder here, as theoretically this could open for the possibility of direct communication between PSP and cardholder in case of payment transaction failure. Not sure about data protection implications of it though.